pkg/user/tx_client.go

```diff
@@ -361,81 +408,83 @@ func (s *TxClient) ConfirmTx(ctx context.Context, txHash string) (*sdktypes.TxRe
 	}
 }
 
-func (s *TxClient) EstimateGas(ctx context.Context, msgs []sdktypes.Msg, opts ...TxOption) (uint64, error) {
-	s.mtx.Lock()
-	defer s.mtx.Unlock()
+func (client *TxClient) EstimateGas(ctx context.Context, msgs []sdktypes.Msg, opts ...TxOption) (uint64, error) {
```

Comment:
nit: maybe we should document that the gas estimate we get is always multiplied by the multiplier in case that's not clear